### ISSUE TITLE
check_requirements.py: No such file or directory

### ISSUE THREAD

**Issue by handsomezebra** [allenai/allennlp#1561]
**Describe the bug**
Getting error `/bin/sh: ./scripts/check_requirements.py: No such file or directory` when I run verify.py.

**To Reproduce**
Steps to reproduce the behavior
1. Clone the latest master
2. `./scripts/verify.py --checks check-requirements`

**System (please complete the following information):**
 - OS: Ubuntu 16.04
 - Python version: Python 3.6.5 :: Anaconda, Inc.
 - AllenNLP version: master
 - PyTorch version: 0.4.0

**Comment by nelson-liu** [comment]
I guess `https://github.com/allenai/allennlp/blob/master/scripts/verify.py#L47` should be changed to `run("./scripts/check_requirements_and_setup", shell=True, check=True)`

**Comment by nelson-liu** [comment]
would you like to submit a PR?

**Comment by handsomezebra** [comment]
Tried to rename the file, then I got the following error:
```
requirements mismatch
in requirements.txt: {'coverage', 'sphinx_rtd_theme', 'codecov', 'pylint==1.8.1', 'mypy==0.521', 'sphinx==1.5.3', 'pytest-cov', 'sphinx-autobuild', 'jupyter', 'pypandoc', 'aiohttp', 'twine==1.11.0'}
in setup.py: set()
```
Need more time for me to figure out how to fix that.

**Comment by nelson-liu** [comment]
So the requirements you list above aren't necessary for running the unit tests --- they're necessary for running CI and stuff like that. 

Not sure what the best solution here is / whether users should be installing those dependencies. @schmmd probably has thoughts, since he was the one who made the change to the `requirements.txt` in the first place?

**Comment by schmmd** [comment]
Interesting.  I'm not familiar with `check_requirements.py`.  I merged `requirements.txt` `requirements_test.txt` in response to https://github.com/allenai/allennlp/issues/1378.

@joelgrus how would you recommend fixing `./scripts/check_requirements.py`?

**Comment by joelgrus** [comment]
I wrote that because setup.py and requirements.txt kept getting out of sync, but I don't think it's run as a precommit anywhere

**Comment by schmmd** [comment]
@joelgrus now that requirements.txt and requirements_test.txt are merged we expect them to be out of sync.  Should we remove `check_requirements.py` or should we revert my recent PR that merged `requirements.txt` and ` requirements_test.txt`.

**Comment by joelgrus** [comment]
let's just remove it as a default check for verify.py, I still want to run it e.g. before doing a release.

maybe there's a good way to handle the expected discrepancy, but it's not high on my list to figure out.

**Comment by schmmd** [comment]
See https://github.com/allenai/allennlp/pull/1565

**Comment by DeNeutoy** [comment]
@joelgrus Seems like there's a really easy way - just add a commented separator to our reqirements like `############` and then anything above it must be in `setup.py`.

**Comment by schmmd** [comment]
@DeNeutoy that's a good idea.  Mine was to check that `setup.py` is a subset of `requirements.txt`.

**Comment by nelson-liu** [comment]
should we close, since it seems like a fix was made in #1565 ?

**Comment by schmmd** [comment]
Closing--but I opened https://github.com/allenai/allennlp/issues/1571.
